git/filter_process_scanner.go

```diff
@@ -98,25 +98,25 @@ func (o *FilterProcessScanner) Init() error {
 // capabilities given to LFS by the parent, an error will be returned. If there
 // was an error reading or writing capabilities between the two, an error will
 // be returned.
-func (o *FilterProcessScanner) NegotiateCapabilities() error {
+func (o *FilterProcessScanner) NegotiateCapabilities() ([]string, error) {
 	reqCaps := []string{"capability=clean", "capability=smudge"}
 
 	supCaps, err := o.pl.readPacketList()
 	if err != nil {
-		return fmt.Errorf("reading filter-process capabilities failed with %s", err)
+		return nil, fmt.Errorf("reading filter-process capabilities failed with %s", err)
 	}
 	for _, reqCap := range reqCaps {
 		if !isStringInSlice(supCaps, reqCap) {
-			return fmt.Errorf("filter '%s' not supported (your Git supports: %s)", reqCap, supCaps)
+			return nil, fmt.Errorf("filter '%s' not supported (your Git supports: %s)", reqCap, supCaps)
 		}
 	}
 
 	err = o.pl.writePacketList(reqCaps)
```

Comment:
Since we're not sending the `capability=delay` entry here yet, would this change to:

```go
if supCaps.Has("delay") {
        reqCaps = append(reqCaps, "capability=delay")
}

err = o.pl.writePacketList(reqCaps)
```

Since this pull request only sends `capability=clean` and `capability=smudge`, I think this should be fine.